### ISSUE TITLE
blazor.boot.json is 404-ing after AOT deploy

### ISSUE THREAD

**Issue by StachuDotNet** [darklang/dark#3872]
When I go to `use-blazor=true` after #3831 being merged, I get a 404 on blazor.boot.json (specifically, `https://static.darklang.com/blazor/blazor.boot-4cd2df25689084dcc6338095e0f86de9df2593021bc39d6df3884666319f787e.json`)

Either the hash there is incorrect (a blazor.boot.json exists, but not at that location), or no blazor.boot.json is there for some reason. Currently investigating.

**Comment by StachuDotNet** [comment]
I've looked through `.circleci/config.yml`, and I'm guessing this is some sort of problem with our `_push-assets-to-cdn` script?
How can I see the static assets in production? @pbiggar or maybe that's not the right question.

**Comment by StachuDotNet** [comment]
When I check my local `etags.json`, I see that `BlazorWorker.js` has `2770c79104479e17e0b4f140b571406e9cc93a495e0da0926a0832ee6d7cf949` as the value.
That file loads fine in prod currently.

`blazor/blazor.boot.json`'s value is `92e1133126671a4cf32aca593d3d156f53055dca2e92983ffd2977e312a33701`.

For some reason, `ui.html` is attempting to load `https://static.darklang.com/blazor/blazor.boot-4cd2df25689084dcc6338095e0f86de9df2593021bc39d6df3884666319f787e.json` rather than `https://static.darklang.com/blazor/blazor.boot-92e1133126671a4cf32aca593d3d156f53055dca2e92983ffd2977e312a33701.json` which surprises me.

Note: `https://static.darklang.com/blazor/blazor.boot-92e1133126671a4cf32aca593d3d156f53055dca2e92983ffd2977e312a33701.json` also 404s.

**Comment by pbiggar** [comment]
Here's where they're uploaded: https://app.circleci.com/pipelines/github/darklang/dark/3038/workflows/75a662c9-edf4-4331-940c-c5f1d887a4bb/jobs/32702

I note - and have previously seen in other builds - the `[Killed]` output in there. Possibly related?

**Comment by StachuDotNet** [comment]
the `Killed`s definitely could be related. :thinking:

**Comment by pbiggar** [comment]
Good idea. They might be there from a cache - possibly bump the cache number?

**Comment by pbiggar** [comment]
https://github.com/darklang/dark/blob/main/.circleci/config.yml#L318
https://github.com/darklang/dark/blob/main/.circleci/config.yml#L344

**Comment by StachuDotNet** [comment]
I _roughly_ understand the effect of increasing that to `v7`, but not fully grasping. Could you please shed some light?

I understand how they'd effect CI, but my curiosity was more around why I'm seeing the .br files even _locally_, since `BlazorEnableCompression` is set to `false` in `Wasm.fsproj`

**Comment by pbiggar** [comment]
> I _roughly_ understand the effect of increasing that to `v7`, but not fully grasping. Could you please shed some light?

We're using v6 now, so there are some v6 caches, which load something that was built some time ago.

If we change it to v7, then there are no existing caches, so we start from scratch. So it's equivalent to clearing the caches.
 
> That is, I roughly understand how they'd effect CI, but my curiosity was more around why I'm seeing the .br files even _locally_, since `BlazorEnableCompression` is set to `false` in `Wasm.fsproj`

oh that is weird.

**Comment by StachuDotNet** [comment]
created #3877 - worth a shot.

**Comment by StachuDotNet** [comment]
From #3877, https://app.circleci.com/pipelines/github/darklang/dark/3047/workflows/d7f54802-b5d5-416d-9f5a-92e3d94d5777/jobs/32807/parallel-runs/0/steps/0-117

Still seeing `Killed` after increasing cache #, though seemingly fewer. 

I'm going to add another PR that adds much more logging to that script - I don't find the mimetimes alone very useful as output, and don't see harm in adding much more context for diagnosing situations like these.

**Comment by StachuDotNet** [comment]
The upload to CDN seems to work for some files, and then fail. It's somewhat random which files make it there. For instance, blazor.boot.json finally made it, but some .dlls haven't yet made it so we're still having load problems.

I suspect that the script is parallelize-ing too much - some sort of throttling there, or temporarily giving up on parallelization, may be what will solve this.

**Comment by StachuDotNet** [comment]
I'm currently working on replacing _push-assets-to-cdn with a python script that groups calls to gsutil by mimetype, and I've hit a snag due to my python ignorance.

In bash:
`filehash="$(jq -r --arg FILE "${basefile}" '.[$FILE]' backend/static/etags.json)"`

My attempt to convert this to python:
```
cmd = f"jq -r --arg FILE \"{basefile}\" './[$FILE]' backend/static/etags.json"
filehash = subprocess.check_output(cmd, encoding="utf-8")
```

The failure I get is `FileNotFoundError: [Errno 2] No such file or directory: 'jq -r --arg FILE "unsupportedBrowser.js" \'./\' backend/static/etags.json'`

Is there anything obviously wrong here?
From my understanding, we're using `--arg FILE "${basefile"}` to define a variable FILE for use within jq, and then use `.[$FILE]` to reference that file. It's a bit unclear why we need that misdirection, and it's tripping me up.

**Comment by StachuDotNet** [comment]
Never mind - brain got stuck in a loop and forgot I could just read etags.json as a dict to process - got past that.

**Comment by StachuDotNet** [comment]
This was resolved with #3984 and some follow-up PRs
